mex/common/models/base.py

```diff
@@ -112,31 +114,33 @@ def _fix_value_listyness_for_field(
 
     @model_validator(mode="before")
     @classmethod
-    def fix_listyness(cls, values: ModelValuesT) -> ModelValuesT:
-        """Adjust the listyness of to-be-parsed values to match the desired shape.
+    def fix_listyness(cls, data: RawModelDataT) -> RawModelDataT:
+        """Adjust the listyness of to-be-parsed data to match the desired shape.
 
-        If the model defines a list[T] field but the raw data contains just a value
-        of type T, it will be wrapped into a list. If the raw data contains a literal
-        `None`, but the list field is defined as required, we substitute an empty list.
+        If that data is a Mapping and the model defines a list[T] field but the raw data
+        contains just a value of type T, it will be wrapped into a list. If the raw
+        data contains a literal `None`, but the list field is defined as required, we
+        substitute an empty list.
 
         If the model does not expect a list, but the raw data contains a list with
         no entries, it will be substituted with `None`. If the raw data contains exactly
         one entry, then it will be unpacked from the list. If it contains more than one
         entry however, an error is raised, because we would not know which to choose.
 
         Args:
-            values: Raw values to be parsed
+            data: Raw data to be parsed
 
         Returns:
-            dict: Values with fixed list shapes
+            data with fixed list shapes
         """
-        for name, value in values.items():
-            field_name = cls._get_alias_lookup().get(name, name)
-            if field := cls.model_fields.get(field_name):
-                values[name] = cls._fix_value_listyness_for_field(
-                    field_name, field, value
-                )
-        return values
+        if isinstance(data, MutableMapping):
```

Comment:
tiny test for the else-case would be nice

Reply:
I agree but I fail to create one. The error this adresses is raised in mex-backend during the model validation that fastapi controls: https://github.com/robert-koch-institut/mex-backend/actions/runs/7196464325/job/19601388928
Any idea how to recreate that? I tried to supply a model to model-validate instead of a dictionary, but pydantic seems to convert that to a dict prior to executing fix_listyness.

Reply:
added one in 0e310ad9000e2aee150574f9d377a268eb5eaf95